### PR TITLE
[homework #8] - js-advanced-1 by Edelmiro Anton

### PR DESCRIPTION
**Pull request by EdelmiroAnton**
Hey @ilya-ischenko!
Here's my hw. I'm finishing the other tasks. 
Please let me know if I have to do some modifications.

Thanks!

### CODE REVIEW

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    return (timeAgo = hours + ' hours ' + minutes / 5 + ' minutes ago');
+  }
+  if (hours >= 24) {
+    return (timeAgo = days + wordDay + ' ago');
+  }
+
+  if (days >= 365) {
+    return (timeAgo = days + ' days ago ');
+  }
+}
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:74]: Be aware that you can use new features for strings concatination, for example: `${hours} hours ${minutes / 5} minutes ago` etc...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals
Looks a little bit cleaner as for me

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+// Task 4 - Create a function that generate a random date between to dates
+
+const date_1 = new Date(2021, 0, 23);
+const date_2 = new Date(2021, 1, 23);
+
+function randomDate(firstDate, secondDate) {
+  let date = new Date(firstDate.getTime() + Math.random() * (secondDate.getTime() - firstDate.getTime()));
+  const day = `${date.getDate() < 10 ? '0' : ''}${date.getDate()}`;
+  day;
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:84]: looks like typo

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+  const month = `${date.getMonth() < 10 ? '0' : ''}${date.getMonth() + 1}`;
+  month;
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:87]: looks like typo

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+  const year = `${date.getFullYear()}`;
+  year;
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:90]: looks like typo

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  this.firstName = first;
+  this.lastName = last;
+
+  Object.defineProperty(this, 'fullName', {
+    get: function () {
+      return this.firstName + ' ' + this.lastName;
+    },
+
+    set(value) {
+      value = value.trim();
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:119]: Try to not mutate function arguments, it will be helpful in future

**Review by ilya-ischenko — COMMENTED** [review]
Looks good as for me!

1. Do not see first task. Write to me if you have some troubles with it
2. Ok for now, but for next homework use "const" instead of "let" where it possible

**Comment by EdelmiroAnton** [comment]
Hi @ilya-ischenko! I fix the code and add the task 1. Let me know if now is it ok.

Thanks in advance!

File: homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+  for (i = 0; i < splitted.length; i++) {
+    if (!finalValue[splitted[i]]) {
+      return null;
+    }
+    finalValue = finalValue[splitted[i]];
+  }
+
+  return finalValue;
+}
+
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js:17]: Make sure that this function maintain all needed cases, please

`const randomValue = Math.random();
const nullValue = null;

console.log(pluck(user, 'preferences.sound.value')); // 30
console.log(pluck(user, 'unknown.key')); // null
console.log(pluck(randomValue, 'unknown.key')); // null
console.log(pluck(nullValue, 'unknown.key')); // null`

Comment (EdelmiroAnton) [reply homeworks/edelmiro-anton/08-js-advanced-1/advanced-1.js]: I just fix it. I add the code for cases where is not an object return null.
Thanks!

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**